src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,46 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add phone to DB " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            List<Predicate> predicates = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<String> predicateOfParameter
+                        = criteriaBuilder.in(phoneRoot.get(entry.getKey()));
+                for (String parameter : entry.getValue()) {
+                    predicateOfParameter.value(parameter);
+                }
+                predicates.add(predicateOfParameter);
+            }
+            query.where(criteriaBuilder.and(predicates.toArray(Predicate[]::new)));
```

Comment:
You can try to pass just an array of predicates (predicates.toArray(Predicate[]::new)) into the where() method without using criteriaBuilder.and(). It should work